src/main/java/seedu/duke/InternshipList.java

```diff
@@ -26,9 +26,16 @@ public InternshipList(ArrayList<Internship> internships) {
     }
 
     public void addInternship(Internship internship) {
+
+        assert internship != null : "Internship object cannot be null";
+
         if (internship.getId() == -1) {
             internship.setId(internships.size());
         }
+
+        assert internship.getId() >= 0 : "Internship ID should be non-negative after setting";
+        assert internship.getId() == internships.size() : "Internship ID should match the list size before adding";
```

Comment:
could this assert line be changed to :
assert internship.getid() == internships.size() + 1  
Since internship.setId(internships.size()) sets the id to be internships.size() + 1. 

The relevant code snippet for the setId() function has been attached below
![Screenshot 2024-11-03 155523](https://github.com/user-attachments/assets/2ad9c5a3-8aa7-4d70-9ea9-f5725453ba5b)

Reply:
thanks for pointing that out! changes made worked, sorry for the error!